ext/nnstreamer/tensor_filter/tensor_filter_caffe2_core.cc

```diff
@@ -0,0 +1,465 @@
+        inputTensorMeta.info[i].dimension[1],\
+        inputTensorMeta.info[i].dimension[2],\
+        inputTensorMeta.info[i].dimension[3]\
+      },\
+      at::dtype<type> ().device (CPU)\
+  );\
+} while (0);
+
+/**
+ * @brief initialize the input tensor
+ */
+int
+Caffe2Core::initInputTensor ()
+{
+  int i = 0;
+  inputTensorMap.clear ();
+  for (; i < inputTensorMeta.num_tensors; i++) {
+    Tensor *inputTensor = workSpace.CreateBlob (inputTensorMeta.info[i].name)
+      ->GetMutable<Tensor> ();
+
+  switch (inputTensorMeta.info[i].type){
+      case _NNS_INT32:
+        initializeTensor (int32_t);
+        break;
+      case _NNS_UINT32:
+        g_critical ("invalid data type is used");
+        return -1;
+      case _NNS_INT16:
+        initializeTensor (int16_t);
+        break;
+      case _NNS_UINT16:
+        initializeTensor (uint16_t);
+        break;
+      case _NNS_INT8:
+        initializeTensor (int8_t);
+        break;
+      case _NNS_UINT8:
+        initializeTensor (uint8_t);
+        break;
+      case _NNS_FLOAT64:
+        initializeTensor (double);
+        break;
+      case _NNS_FLOAT32:
+        initializeTensor (float);
+        break;
+      case _NNS_INT64:
+        initializeTensor (int64_t);
+        break;
+      case _NNS_UINT64:
+        g_critical ("invalid data type is used");
+        return -1;
+      default:
+        g_critical ("invalid data type is used");
+        return -1;
+    }
+
+    inputTensorMap.insert (
+      std::make_pair (inputTensorMeta.info[i].name, inputTensor)
+    );
+  }
+  return 0;
+}
+
+/**
+ * @brief	get the model path
+ * @return the model path.
+ */
+const char *
+Caffe2Core::getPredModelPath ()
+{
+  return pred_model_path;
+}
+
+/**
+ * @brief	get the model path
+ * @return the model path.
+ */
+const char *
+Caffe2Core::getInitModelPath ()
+{
+  return init_model_path;
+}
+
+/**
+ * @brief	load the caffe2 model
+ * @note	the model will be loaded
+ * @return 0 if OK. non-zero if error.
+ */
+int
+Caffe2Core::loadModels ()
+{
+#if (DBG)
+  gint64 start_time = g_get_real_time ();
+#endif
+  if (!g_file_test (init_model_path, G_FILE_TEST_IS_REGULAR)) {
+    g_critical ("the file of init_model_path is not valid\n");
+    return -1;
+  }
+  if (!g_file_test (pred_model_path, G_FILE_TEST_IS_REGULAR)) {
+    g_critical ("the file of pred_model_path is not valid\n");
+    return -1;
+  }
+
+  CAFFE_ENFORCE (ReadProtoFromFile (init_model_path, &initNet));
+  CAFFE_ENFORCE (ReadProtoFromFile (pred_model_path, &predictNet));
+
+  CAFFE_ENFORCE (workSpace.RunNetOnce (initNet));
+  CAFFE_ENFORCE (workSpace.CreateNet (predictNet));
+#if (DBG)
+  gint64 stop_time = g_get_real_time ();
+  g_message ("Model is loaded: %" G_GINT64_FORMAT, (stop_time - start_time));
+#endif
+  return 0;
+}
+
+/**
+ * @brief extract and store the information of input tensors
+ * @return 0 if OK. non-zero if error.
+ */
+int
+Caffe2Core::setInputTensorProp (const GstTensorFilterProperties * prop)
```

Comment:
IMO setInputTensorProp() and setOutputTensorProp() these functions are unnecessary. 
plz consider to copy the meta data in init function.

Reply:
it's removed